packages/flutter_tools/test/commands/packages_test.dart

```diff
@@ -120,20 +145,67 @@ class MockProcessManager implements ProcessManager {
     bool runInShell: false,
     ProcessStartMode mode: ProcessStartMode.NORMAL,
   }) {
-    onStart(command);
-    return new Future<Process>.value(new MockProcess());
+    return new Future<Process>.value(processFactory(command));
   }
 
   @override
   dynamic noSuchMethod(Invocation invocation) => null;
 }
 
+/// Process that prompts the user to proceed, then asynchronously writes
+/// some lines to stdout before it exits.
+class PromptingProcess implements Process {
+  PromptingProcess(String prompt, List<String> outputLines) {
```

Comment:
To me this seems overly general for a one-off test. Perhaps hardcode the prompt and output lines - I think that would make the test easier to read.

Reply:
I didn't do this, since the prompt and the output lines are really part of the test data, so they belong in the test itself. But the code has been restructured to keep test logic and data closer. PTAL.